### PR TITLE
[POAE7-2743]update protobuf to 3.21.4 and install nasm

### PR DESCRIPTION
**Pull request by PenghuiJiao**
please take review @xieqi @Ma-Jian1 
### What changes were proposed in this pull request?

update protobuf to 3.21.4 and install nasm

### Why are the changes needed?

update protobuf to 3.21.4 and install nasm

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

CICD
### Which label does this PR belong to?

CICD

### CODE REVIEW

File: ci/docker/Dockerfile [hunk 1]
```diff
@@ -21,6 +21,7 @@ RUN apt-get update && \
     zip \
     ca-certificates \
     python3-sphinx \
+    nasm \
```
Comment (jikunshang) [inline ci/docker/Dockerfile:24]: why need this?

**Comment by jikunshang** [comment]
BTW, do we need update gcc version?

File: ci/docker/Dockerfile (hunk 1 above)
Comment (xieqi) [reply ci/docker/Dockerfile:24]: build ICL need nasm

**Review by xieqi — APPROVED** [review]

**Review by harborn — APPROVED** [review]

**Comment by jikunshang** [comment]
merged. Lint issue should have been fixed in other PRs.

**Pull request merged**